server/tests/db/test_connection_pool.py

```
@@ -254,6 +254,183 @@ def test_putconn_failure_swallowed(self, fresh_pool):
+        must not be left on a connection that goes back to the pool.
+        """
+        pool, factory = fresh_pool
+        connection, cursor = _make_conn()
+        # First execute (SET user_id) raises; subsequent calls (RESET) succeed.
+        cursor.execute.side_effect = [Exception("db gone"), None, None]
+        factory.return_value.getconn.return_value = connection
+
+        with flask_app.test_request_context(
+            "/api/x", headers={"X-User-ID": "u-1", "X-Org-ID": "org-7"},
+        ):
+            with pool.get_connection() as conn:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)